nsxt/resource_nsxt_policy_context_profile.go

```diff
@@ -0,0 +1,506 @@
+}
+
+var subAttributeKeyMap = map[string]string{
+	"tls_cipher_suite": model.PolicySubAttributes_KEY_TLS_CIPHER_SUITE,
+	"tls_version":      model.PolicySubAttributes_KEY_TLS_VERSION,
+	"cifs_smb_version": model.PolicySubAttributes_KEY_CIFS_SMB_VERSION,
+}
+
+var subAttributeReverseKeyMap = map[string]string{
+	model.PolicySubAttributes_KEY_TLS_CIPHER_SUITE: "tls_cipher_suite",
+	model.PolicySubAttributes_KEY_TLS_VERSION:      "tls_version",
+	model.PolicySubAttributes_KEY_CIFS_SMB_VERSION: "cifs_smb_version",
+}
+
+func resourceNsxtPolicyContextProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyContextProfileCreate,
+		Read:   resourceNsxtPolicyContextProfileRead,
+		Update: resourceNsxtPolicyContextProfileUpdate,
+		Delete: resourceNsxtPolicyContextProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"app_id":       getContextProfilePolicyAppIDAttributesSchema(),
+			"domain_name":  getContextProfilePolicyOtherAttributesSchema(),
+			"url_category": getContextProfilePolicyOtherAttributesSchema(),
+		},
+	}
+}
+
+func getContextProfilePolicyAppIDAttributesSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"description": getDescriptionSchema(),
+				"is_alg_type": {
+					Type:        schema.TypeBool,
+					Description: "Whether the app_id value is ALG type or not",
+					Computed:    true,
+				},
+				"value": {
+					Type:        schema.TypeSet,
+					Description: "Values for attribute key",
+					MinItems:    1,
+					Required:    true,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+				},
+				"sub_attribute": getPolicyAttributeSubAttributeSchema(),
+			},
+		},
+	}
+}
+
+func getContextProfilePolicyOtherAttributesSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"description": getDescriptionSchema(),
+				"value": {
+					Type:        schema.TypeSet,
+					Description: "Values for attribute key",
+					Required:    true,
+					MinItems:    1,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+				},
+				"sub_attribute": getPolicyAttributeSubAttributeSchema(),
+			},
+		},
+	}
+}
+
+func getPolicyAttributeSubAttributeSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeSet,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"tls_cipher_suite": getPolicyAttributeSubAttributeValueSchema("tls_cipher_suite"),
```

Comment:
are these the only possible sub-attrs today?

Reply:
Yes as documented in API doc